### PR TITLE
Add CandumpReader for rdcandump

### PR DESCRIPTION
**Pull request by polybassa**

### CODE REVIEW

**Comment by polybassa** [comment]
Hi p-l-, I didn't got your question completely. Do you mean, if I could implement a `CandumpReader` class, which is used by `rdcandump`? To follow the same way of implementation.

**Comment by polybassa** [comment]
I looked into the implementation of `PcapReader`. I will do the same for `candump`.

**Comment by polybassa** [comment]
Done. Maybe the `metaclass` is not necessary, but I was not sure about that. What do you think?

**Comment by gpotter2** [comment]
If you ask me, `ISOTPSniffer` could be rewritten using `AsyncSniffer`

File: scapy/contrib/isotp.py [hunk 1]
```diff
@@ -465,22 +466,26 @@ def __init__(self):
...
             while not c.stop and m.count() > 0:
                 rcvd = m.pop()
+                print("Popped ", repr(rcvd))
                 on_pkt(rcvd)
 
         def internal_stop_filter(p):
             return c.stop
 
         def on_pkt(p):
+            print(repr(p))
```
Comment (gpotter2) [inline scapy/contrib/isotp.py]: Apart from that those are probably debug comments

Comment (polybassa) [reply scapy/contrib/isotp.py]: Fixed

**Comment by polybassa** [comment]
I had to modify the `DefaultSession`. Can you please check if that modification is ok.

**Comment by gpotter2** [comment]
It's alright

**Comment by polybassa** [comment]
I’m thinking about the ‘count’ argument of sniff. Wouldn’t it be more intuitive to let the session count the packets? A ISOTP message can consist out of many CAN frames. The number of CAN packtest isn’t related to the number of ISOTP frames, which would let sniff return a “random” number of isotp packets if a count argument is provided. I think the same applies for the IPSession. Would do you think? Should that behavior be changed?

**Comment by gpotter2** [comment]
It makes sense, you are right. With `TCPSession` (for instance HTTP), this might get even more problematic...

If you want to do that in this PR, feel free to. Otherwise I'll probably do it myself.

**Comment by polybassa** [comment]
Done

**Pull request merged**